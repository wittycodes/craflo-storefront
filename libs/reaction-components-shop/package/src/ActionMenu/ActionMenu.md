### Overview

The Catalyst ActionMenu is based on the the Material-UI example for [Menus](https://material-ui.com/components/menus/). It uses the Catalyst [Button](https://catalyst.reactioncommerce.com/#/Components/Actions/Button) and the Material-UI [Menu](https://material-ui.com/api/menu/) components.

### Usage

#### Simple ActionMenu

This is basic example of how to implement the ActionMenu.

```jsx
const options = [{
  label: "Open"
}, {
  label: "Archived"
}, {
  label: "Shipped"
}, {
  label: "Canceled"
} ];

<ActionMenu
  options={options}
  onSelect={(option, index) => alert(`Selected option "${option.label}" at index (${index})`)}
>
  Set status
</ActionMenu>
```

#### Options with confirmation

ActionMenu options can be guarded with a confirmation dialog on selection. To achieve this, provide any combination of `confirmTitle` and `confirmMessage` as part of the option object. You may also supply an `onClick` handler to each option for more control of the "click" action. The `onSelect` callback will still fire when an item is selected. In most cases you'll want to avoid using `onSelect` and `onClick` for each option together unless you plan to do some advanced event handling using both.

```jsx
const options = [{
  label: "Filter by file",
  onClick: () => {
    console.log("Filter by file");
  }
}, {
  label: "Publish",
  confirmTitle: "Publish 32 products",
  confirmMessage: "Are you sure you want to publish 32 products to your storefront?",
  onClick: () => {
    console.log("Published 32 products");
  }
}, {
  label: "Make Visible",
  confirmTitle: "Make 32 products visible",
  confirmMessage: "Are you sure you want to make 32 products visible to customers?",
  onClick: () => {
    console.log("Made 32 products visible");
  }
}, {
  label: "Make Hidden",
  confirmTitle: "Make 32 products hidden",
  confirmMessage: "Are you sure you want to make 32 products hidden from customers?",
  onClick: () => {
    console.log("Made 32 products hidden");
  }
}, {
  label: "Duplicate",
  confirmTitle: "Duplicate 32 products",
  confirmMessage: "Are you sure you want to duplicate 32 products?",
  onClick: () => {
    console.log("Duplicated 32 products");
  }
}, {
  label: "Archive",
  confirmTitle: "Archive 32 products",
  confirmMessage: "Are you sure you want to archive 32 products? This will hide them from both admins and customers.",
  onClick: () => {
    console.log("Archived 32 products");
  }
}];

<ActionMenu options={options}>
  Actions
</ActionMenu>
```

#### Add detail text to options

Add a second line of text to the menu options to provide additional context to an action.

```jsx
const options = [{
  label: "Add tags to products",
  details: "Add the selected tags from the filtered products list provided."
}, {
  label: "Remove tags from products",
  details: "Remove the selected tags from the filtered products list provided."
}, {
  label: "Remove all tags from products",
  details: "Remove all tags from the filtered products list provided.",
  isDisabled: true
}];

<ActionMenu
  options={options}
  onSelect={(option, index) => alert(`Selected option "${option.label}" at index (${index})`)}
>
  Actions
</ActionMenu>
```

#### Other variations

```jsx
import { Grid } from "@material-ui/core";

const options = [{
  label: "Add tags to products",
  details: "Add the selected tags from the filtered products list provided."
}, {
  label: "Remove tags from products",
  details: "Remove the selected tags from the filtered products list provided."
}, {
  label: "Remove all tags from products",
  details: "Remove all tags from the filtered products list provided.",
  isDisabled: true
}];

const onSelect = (option, index) => alert(`Selected option "${option.label}" at index (${index})`);

<Grid container spacing={2}>
  <Grid item>
    <ActionMenu
      options={options}
      variant="contained"
      onSelect={onSelect}
    >
      Actions
    </ActionMenu>
  </Grid>
  <Grid item>
    <ActionMenu
      options={options}
      color="secondary"
      variant="contained"
      onSelect={onSelect}
    >
      Actions
    </ActionMenu>
  </Grid>
  <Grid item>
    <ActionMenu
      options={options}
      color="error"
      variant="contained"
      onSelect={onSelect}
    >
      Actions
    </ActionMenu>
  </Grid>
  <Grid item>
    <ActionMenu
      options={options}
      color="secondary"
      onSelect={onSelect}
    >
      Actions
    </ActionMenu>
  </Grid>
  <Grid item>
    <ActionMenu
      options={options}
      color="error"
      onSelect={onSelect}
    >
      Actions
    </ActionMenu>
  </Grid>
  <Grid item>
    <ActionMenu
      options={options}
      variant="text"
      onSelect={onSelect}
    >
      Actions
    </ActionMenu>
  </Grid>
</Grid>
```