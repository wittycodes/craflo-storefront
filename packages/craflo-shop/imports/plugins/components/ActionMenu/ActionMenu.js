"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _core = require("@material-ui/core");

var _ChevronDown = _interopRequireDefault(require("mdi-material-ui/ChevronDown"));

var _Button = _interopRequireDefault(require("../Button"));

var _ConfirmDialog = _interopRequireDefault(require("../ConfirmDialog"));

var useStyles = (0, _core.makeStyles)(function (theme) {
  return {
    button: {
      paddingRight: theme.spacing(1.5)
    }
  };
});
/**
 * @name ActionMenu
 * @param {Object} props Component props
 * @returns {React.Component} A React component
 */

var ActionMenu = _react.default.forwardRef(function ActionMenu(props, ref) {
  var children = props.children,
      onSelect = props.onSelect,
      options = props.options,
      otherProps = (0, _objectWithoutProperties2.default)(props, ["children", "onSelect", "options"]);
  var classes = useStyles();

  var _React$useState = _react.default.useState(false),
      _React$useState2 = (0, _slicedToArray2.default)(_React$useState, 2),
      open = _React$useState2[0],
      setOpen = _React$useState2[1];

  var anchorRef = ref || _react.default.useRef(null);
  /**
   * Handle menu item click
   * @param {SyntheticEvent} event Event object
   * @param {Number} index Menu item index
   * @returns {undefined}
   */


  function handleMenuItemClick(_ref) {
    var event = _ref.event,
        index = _ref.index,
        onClick = _ref.onClick;
    var selectedOption = options[index];
    onSelect && onSelect(selectedOption, index);
    onClick && onClick(event);
    setOpen(false);
  }
  /**
   * Toggle menu open
   * @returns {undefined}
   */


  function handleToggle() {
    setOpen(function (prevOpen) {
      return !prevOpen;
    });
  }
  /**
   * Handle menu close
   * @param {SyntheticEvent} event Event object
   * @returns {undefined}
   */


  function handleClose(event) {
    if (anchorRef.current && anchorRef.current.contains(event.target)) {
      return;
    }

    setOpen(false);
  }

  return /*#__PURE__*/_react.default.createElement(_react.Fragment, null, /*#__PURE__*/_react.default.createElement(_Button.default, (0, _extends2.default)({
    "aria-controls": "action-menu",
    "aria-haspopup": "true",
    className: classes.button,
    onClick: handleToggle,
    ref: anchorRef
  }, otherProps), children, /*#__PURE__*/_react.default.createElement(_core.Box, {
    display: "flex",
    paddingLeft: 1
  }, /*#__PURE__*/_react.default.createElement(_ChevronDown.default, null))), /*#__PURE__*/_react.default.createElement(_core.Menu, {
    MenuListProps: {
      disablePadding: true
    },
    anchorEl: anchorRef.current,
    id: "action-menu",
    keepMounted: true,
    open: open,
    onClose: handleClose
  }, /*#__PURE__*/_react.default.createElement(_core.MenuItem, {
    key: "default-label",
    disabled: true
  }, /*#__PURE__*/_react.default.createElement(_core.Box, {
    maxWidth: 320,
    whiteSpace: "normal"
  }, /*#__PURE__*/_react.default.createElement(_core.ListItemText, {
    primary: children
  }))), options.map(function (option, index) {
    var label = option.label,
        details = option.details,
        isDisabled = option.isDisabled,
        cancelActionText = option.cancelActionText,
        confirmActionText = option.confirmActionText,
        confirmTitle = option.confirmTitle,
        confirmMessage = option.confirmMessage,
        onClick = option.onClick;

    var callback = function callback(event) {
      return handleMenuItemClick({
        event: event,
        index: index,
        onClick: onClick
      });
    };

    if (confirmTitle || confirmMessage) {
      return /*#__PURE__*/_react.default.createElement(_ConfirmDialog.default, {
        key: "dialog-".concat(index),
        cancelActionText: cancelActionText,
        confirmActionText: confirmActionText,
        title: confirmTitle,
        message: confirmMessage,
        onConfirm: callback
      }, function (_ref2) {
        var openDialog = _ref2.openDialog;
        return /*#__PURE__*/_react.default.createElement(_core.MenuItem, {
          key: index,
          disabled: isDisabled,
          onClick: openDialog
        }, /*#__PURE__*/_react.default.createElement(_core.Box, {
          maxWidth: 320,
          whiteSpace: "normal"
        }, /*#__PURE__*/_react.default.createElement(_core.ListItemText, {
          primary: label,
          secondary: details
        })));
      });
    }

    return /*#__PURE__*/_react.default.createElement(_core.MenuItem, {
      key: index,
      disabled: isDisabled,
      onClick: callback
    }, /*#__PURE__*/_react.default.createElement(_core.Box, {
      maxWidth: 320,
      whiteSpace: "normal"
    }, /*#__PURE__*/_react.default.createElement(_core.ListItemText, {
      primary: label,
      secondary: details
    })));
  })));
});

ActionMenu.defaultProps = {
  color: "primary",
  variant: "outlined"
};
ActionMenu.propTypes = {
  /**
   * The content of the Button
   */
  children: _propTypes.default.node,

  /**
   * Override or extend the styles applied to the component.
   */
  classes: _propTypes.default.object,

  /**
   * Options: `default` | `inherit` | `primary` | `secondary` | `error`
   */
  color: _propTypes.default.string,

  /**
   * If `true`, the button will be disabled.
   */
  disabled: _propTypes.default.bool,
  // eslint-disable-line

  /**
   * If `true`, the CircularProgress will be displayed and the button will be disabled.
   */
  isWaiting: _propTypes.default.bool,

  /**
   * Called when an option is selected. Can be use simultaneously with option onClick callbacks.
   */
  onSelect: _propTypes.default.func,

  /**
   * Menu options
   */
  options: _propTypes.default.arrayOf(_propTypes.default.shape({
    /**
     * Change the cancel button label in the confirm dialog
     */
    cancelActionText: _propTypes.default.string,

    /**
     * Change the label of the confirmation button in the confirm dialog
     */
    confirmActionText: _propTypes.default.string,

    /**
     * If supplied, the option will show a confirm dialog this message when selected.
     */
    confirmMessage: _propTypes.default.string,

    /**
     * If supplied, the option will show a confirm dialog this title when selected
     */
    confirmTitle: _propTypes.default.string,

    /**
     * Secondary option label
     */
    details: _propTypes.default.string,

    /**
     * Disable the option
     */
    isDisabled: _propTypes.default.bool,

    /**
     * Option label
     */
    label: _propTypes.default.string.isRequired,

    /**
     * If supplied, this function will be called in addition to onSelect
     */
    onClick: _propTypes.default.func
  }))
};
var _default = ActionMenu;
exports.default = _default;