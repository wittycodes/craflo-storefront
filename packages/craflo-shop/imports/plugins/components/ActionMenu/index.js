"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _ActionMenu.default;
  }
});

var _ActionMenu = _interopRequireDefault(require("./ActionMenu"));