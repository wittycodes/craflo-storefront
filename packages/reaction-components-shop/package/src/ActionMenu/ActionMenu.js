import React, { Fragment } from "react";
import PropTypes from "prop-types";
import {
  Box,
  ListItemText,
  Menu,
  MenuItem,
  makeStyles
} from "@material-ui/core";
import ChevronDownIcon from "mdi-material-ui/ChevronDown";
import Button from "../Button";
import ConfirmDialog from "../ConfirmDialog";

const useStyles = makeStyles((theme) => ({
  button: {
    paddingRight: theme.spacing(1.5)
  }
}));

/**
 * @name ActionMenu
 * @param {Object} props Component props
 * @returns {React.Component} A React component
 */
const ActionMenu = React.forwardRef(function ActionMenu(props, ref) {
  const {
    children,
    onSelect,
    options,
    ...otherProps
  } = props;
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);
  const anchorRef = ref || React.useRef(null);

  /**
   * Handle menu item click
   * @param {SyntheticEvent} event Event object
   * @param {Number} index Menu item index
   * @returns {undefined}
   */
  function handleMenuItemClick({ event, index, onClick }) {
    const selectedOption = options[index];
    onSelect && onSelect(selectedOption, index);
    onClick && onClick(event);
    setOpen(false);
  }

  /**
   * Toggle menu open
   * @returns {undefined}
   */
  function handleToggle() {
    setOpen((prevOpen) => !prevOpen);
  }

  /**
   * Handle menu close
   * @param {SyntheticEvent} event Event object
   * @returns {undefined}
   */
  function handleClose(event) {
    if (anchorRef.current && anchorRef.current.contains(event.target)) {
      return;
    }

    setOpen(false);
  }

  return (
    <Fragment>
      <Button
        aria-controls="action-menu"
        aria-haspopup="true"
        className={classes.button}
        onClick={handleToggle}
        ref={anchorRef}
        {...otherProps}
      >
        {children}
        <Box display="flex" paddingLeft={1}>
          <ChevronDownIcon />
        </Box>
      </Button>
      <Menu
        MenuListProps={{ disablePadding: true }}
        anchorEl={anchorRef.current}
        id="action-menu"
        keepMounted
        open={open}
        onClose={handleClose}
      >
        <MenuItem key="default-label" disabled>
          <Box maxWidth={320} whiteSpace="normal">
            <ListItemText
              primary={children}
            />
          </Box>
        </MenuItem>
        {options.map((option, index) => {
          const {
            label,
            details,
            isDisabled,
            cancelActionText,
            confirmActionText,
            confirmTitle,
            confirmMessage,
            onClick
          } = option;

          const callback = (event) => handleMenuItemClick({ event, index, onClick });

          if (confirmTitle || confirmMessage) {
            return (
              <ConfirmDialog
                key={`dialog-${index}`}
                cancelActionText={cancelActionText}
                confirmActionText={confirmActionText}
                title={confirmTitle}
                message={confirmMessage}
                onConfirm={callback}
              >
                {({ openDialog }) => (
                  <MenuItem
                    key={index}
                    disabled={isDisabled}
                    onClick={openDialog}
                  >
                    <Box maxWidth={320} whiteSpace="normal">
                      <ListItemText
                        primary={label}
                        secondary={details}
                      />
                    </Box>
                  </MenuItem>
                )}
              </ConfirmDialog>
            );
          }

          return (
            <MenuItem
              key={index}
              disabled={isDisabled}
              onClick={callback}
            >
              <Box maxWidth={320} whiteSpace="normal">
                <ListItemText
                  primary={label}
                  secondary={details}
                />
              </Box>
            </MenuItem>
          );
        })}
      </Menu>
    </Fragment>
  );
});

ActionMenu.defaultProps = {
  color: "primary",
  variant: "outlined"
};

ActionMenu.propTypes = {
  /**
   * The content of the Button
   */
  children: PropTypes.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: PropTypes.object,
  /**
   * Options: `default` | `inherit` | `primary` | `secondary` | `error`
   */
  color: PropTypes.string,
  /**
   * If `true`, the button will be disabled.
   */
  disabled: PropTypes.bool, // eslint-disable-line
  /**
   * If `true`, the CircularProgress will be displayed and the button will be disabled.
   */
  isWaiting: PropTypes.bool,
  /**
   * Called when an option is selected. Can be use simultaneously with option onClick callbacks.
   */
  onSelect: PropTypes.func,
  /**
   * Menu options
   */
  options: PropTypes.arrayOf(PropTypes.shape({
    /**
     * Change the cancel button label in the confirm dialog
     */
    cancelActionText: PropTypes.string,
    /**
     * Change the label of the confirmation button in the confirm dialog
     */
    confirmActionText: PropTypes.string,
    /**
     * If supplied, the option will show a confirm dialog this message when selected.
     */
    confirmMessage: PropTypes.string,
    /**
     * If supplied, the option will show a confirm dialog this title when selected
     */
    confirmTitle: PropTypes.string,
    /**
     * Secondary option label
     */
    details: PropTypes.string,
    /**
     * Disable the option
     */
    isDisabled: PropTypes.bool,
    /**
     * Option label
     */
    label: PropTypes.string.isRequired,
    /**
     * If supplied, this function will be called in addition to onSelect
     */
    onClick: PropTypes.func
  }))
};

export default ActionMenu;
