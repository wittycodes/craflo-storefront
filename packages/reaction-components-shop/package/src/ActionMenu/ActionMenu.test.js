import React from "react";
import { fireEvent, render, waitForElement } from "../tests/index.js";
import ActionMenu from "./ActionMenu";

const options = [{
  label: "Add tags to products"
}, {
  label: "Remove tags from products"
}, {
  label: "Remove all tags",
  isDisabled: true
}];

test("basic snapshot - only default props", () => {
  const { asFragment } = render(<ActionMenu options={options}>Actions</ActionMenu>);
  expect(asFragment()).toMatchSnapshot();
});

test("select an option", async () => {
  const onSelect = jest.fn();
  const { asFragment, getAllByText, getByText } = render(<ActionMenu onSelect={onSelect} options={options}>Actions</ActionMenu>);
  fireEvent.click(getAllByText("Actions")[0]);
  const removeButton = await waitForElement(() => getByText("Remove tags from products"));
  fireEvent.click(removeButton);
  expect(onSelect).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});

test("option onClick", async () => {
  const onClick = jest.fn();
  const optionsWithAnOnClick = [{
    label: "Add tags to products"
  }, {
    label: "Remove tags from products",
    onClick
  }, {
    label: "Remove all tags",
    isDisabled: true
  }];

  const { asFragment, getAllByText, getByText } = render(<ActionMenu options={optionsWithAnOnClick}>Actions</ActionMenu>);
  fireEvent.click(getAllByText("Actions")[0]);
  const removeButton = await waitForElement(() => getByText("Remove tags from products"));
  fireEvent.click(removeButton);
  expect(onClick).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});

test("option onClick with confirmation", async () => {
  const onClick = jest.fn();
  const optionsWithAnOnClick = [{
    label: "Add tags to products"
  }, {
    label: "Remove tags from products",
    confirmTitle: "Confirm action",
    confirmMessage: "Are you sure you want to do that?",
    onClick
  }, {
    label: "Remove all tags",
    isDisabled: true
  }];

  const { asFragment, getAllByText, getByText } = render(<ActionMenu options={optionsWithAnOnClick}>Actions</ActionMenu>);
  // Open the menu
  fireEvent.click(getAllByText("Actions")[0]);
  // Wait for open, then get the "Remove tags from products" button
  const removeButton = await waitForElement(() => getByText("Remove tags from products"));
  fireEvent.click(removeButton);
  // Wait for dialog to open then click the OK button
  const dialogConfirmButton = await waitForElement(() => getByText("OK"));
  fireEvent.click(dialogConfirmButton);
  expect(onClick).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
