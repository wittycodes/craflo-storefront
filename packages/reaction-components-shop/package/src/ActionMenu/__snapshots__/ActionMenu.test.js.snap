// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot - only default props 1`] = `
<DocumentFragment>
  <button
    aria-controls="action-menu"
    aria-haspopup="true"
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1 MuiButton-outlinedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Actions
      <div
        class="MuiBox-root MuiBox-root-38"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
          />
        </svg>
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`option onClick 1`] = `
<DocumentFragment>
  <button
    aria-controls="action-menu"
    aria-haspopup="true"
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-297 MuiButton-outlinedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Actions
      <div
        class="MuiBox-root MuiBox-root-334"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
          />
        </svg>
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`option onClick with confirmation 1`] = `
<DocumentFragment>
  <button
    aria-controls="action-menu"
    aria-haspopup="true"
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-445 MuiButton-outlinedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Actions
      <div
        class="MuiBox-root MuiBox-root-482"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
          />
        </svg>
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`select an option 1`] = `
<DocumentFragment>
  <button
    aria-controls="action-menu"
    aria-haspopup="true"
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-149 MuiButton-outlinedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Actions
      <div
        class="MuiBox-root MuiBox-root-186"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
          />
        </svg>
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
